toolchain/parse/tree.h

```diff
@@ -106,6 +106,8 @@ class Tree : public Printable<Tree> {
   // Tests whether there are any errors in the parse tree.
   auto has_errors() const -> bool { return has_errors_; }
 
+  auto set_has_errors(bool has_errors) -> void { has_errors_ = has_errors; }
```

Comment:
Might it make sense to call this method `Finalize`, to make explicit that it signifies the point at which `*this` becomes immutable, and should only be called once?

Reply:
For contrast with SemIR, we have SemIR::File::set_has_errors, and Check::Context::Finalize. Here, I'm adding Parse::Tree::Set_has_errors, but it didn't seem worth adding a Parse::Context::Finalize (Check::Context::Finalize also doesn't call SemIR::File::set_has_errors).